src/main/kotlin/org/stellar/walletsdk/util/CreateDecoratedSignature.kt

```diff
@@ -0,0 +1,26 @@
+package org.stellar.walletsdk.util
+
+import java.util.Base64
+import org.stellar.sdk.KeyPair
+import org.stellar.sdk.xdr.DecoratedSignature
+import org.stellar.sdk.xdr.Signature
+
+fun createDecoratedSignature(
+  publicKey: String,
+  signatureBase64String: String,
+  base64Decoder: ((String) -> ByteArray)? = null
+): DecoratedSignature {
+  val signature = Signature()
+
+  if (base64Decoder == null) {
+    signature.signature = Base64.getDecoder().decode(signatureBase64String)
+  } else {
+    signature.signature = base64Decoder(signatureBase64String)
+  }
```

Comment:
Why would someone pass their own `base64Decoder`? It seems too low-level for the wallet SDK.

Reply:
`java.util.Base64` that we're using in Wallet SDK is supported only in Android API 23+. Anything older will need to use `android.util.Base64`. We did that in the [demo](https://github.com/quietbits/android-wallet-demo-test/blob/main/app/src/main/java/com/example/androidwalletdemo/view/RecoverAccount.kt#L45:L47) since we need to support Android API 21. I thought it was better to offer a fallback than ask users to build decorated signatures themselves. I'll add a note about it in the readme once I add this method, hopefully, that will make it clear enough.

Reply:
Is there a way to detect what android API we're using? Or can we try `java.util.Base64` and if that fails due to lack of support, try `andriod.util.Base64`?

Reply:
We can't use `android.util.Base64` in Kotlin Wallet SDK.

Reply:
Oh I see.